core/src/main/java/io/grpc/ChannelImpl.java

```diff
@@ -356,10 +370,23 @@ public boolean isReady() {
     /**
      * Set missing properties on the headers. The given headers will be mutated.
      * @param headers the headers to complete
+     * @param timeoutMicros timeout in microseconds. -1 for infinite. Otherwise, must &gt; 0
      */
-    private void completeHeaders(Metadata.Headers headers) {
+    private void completeHeaders(Metadata.Headers headers, long timeoutMicros) {
```

Comment:
Maybe just pass `CallOption` to this method and have it do the conversion to micros?